great_expectations/expectations/expectation.py

```diff
@@ -179,25 +191,43 @@ def inner_func(
 def param_method(param_name: str) -> Callable:
     """
     Decorator that wraps helper methods dealing with dynamic attributes on RendererConfiguration.params. Ensures a given
-    param_name exists on RendererConfiguration.params before executing the helper method because params are defined by
-    the renderer.
+    param_name exists and is not None before executing the helper method. Params are unknown as they are defined by the
+    renderer, and if a given param is None, no value was set/found that can be used in the helper method.
+
+    If a helper method is decorated with @param_method(param_name="<param_name>") and the param attribute does not
```

Comment:
This seems like a mistake by the author. Should we `raise` an error in this case?

Reply:
When the decorated `classmethod` is called, the `RendererConfiguration` dynamic param attribute is not guaranteed to exist according to the static type checker (`params` are added at runtime). Whether passing around the `RendererConfiguration` (most simple/elegant) or the relevant `RendererParam` to the `classmethod`, the type checker will still require you to confirm that the `RendererParam` is not `None` somewhere (either in the renderer or the helper method). This decorator simplifies the code in the renderer (no need to check for `None` there) and also ensures that a useful error message is produced by the `classmethod` if the renderer implementer did not use the `add_param` instance method to add the param. The decorated methods are `Expectation` `classmethod`s instead of `RendererConfiguration` instance methods because the helpers exist only for simplifying rendering of common params that may not exist for many Expectations.